test/functional/lua/fs_spec.lua

```diff
@@ -278,6 +280,36 @@ describe('vim.fs', function()
     end)
   end)
 
+  describe('root()', function()
+    before_each(function()
+      command(
+        string.format('edit %s', vim.fs.joinpath('test', 'functional', 'fixtures', 'tty-test.c'))
+      )
+    end)
+
+    it('works with a single marker', function()
+      eq(test_source_path, exec_lua([[return vim.fs.root(0, '.git')]]))
+    end)
+
+    it('works with multiple markers', function()
+      local bufnr = api.nvim_get_current_buf()
+      eq(
+        vim.fs.joinpath(test_source_path, 'test', 'functional', 'fixtures'),
+        exec_lua([[return vim.fs.root(..., {'CMakeLists.txt', '.git'})]], bufnr)
+      )
+    end)
+
+    it('works with a function', function()
+      ---@type string
+      local result = exec_lua([[
+        return vim.fs.root(0, function(name, path)
+          return name:match('%.txt$')
+        end)
+      ]])
+      eq(vim.fs.joinpath(test_source_path, 'test', 'functional', 'fixtures'), result)
```

Comment:
this might be a nit, but i've seen this proliferate in other projects. why use join() on literals?
```suggestion
      eq(vim.fs.joinpath(test_source_path, 'test/functional/fixtures'), result)
```

Reply:
Just habit honestly.

Reply:
ffc16b6b5ddb5f1ee59566b2069db953462f06dc